packages/chat/components/chat/src/chat.ts

```diff
@@ -0,0 +1,355 @@
+      response = await LlamaPluginAPI.sendMessageWatsonX(
+        this.apiURL,
+        this.model,
+        this.temperature,
+        this.userprompt,
+        this._messages,
+        searchQuery,
+        this._session,
+        this._eventNumber
+      );
+    }
+    return response;
+  }
+
+
+  /** handle user inputs inside the input field, trigger a search upon an 'enter' key down event
+   * @param{event} lit event sent by the the text input object within the chat
+   **/
+  _handleInput(event) {
+    const { value } = event.target;
+    this._inputText = value;
+    if(event.key == 'Enter') {
+      this._sendInput()
+    }
+  }
+
+  /** handle test selection to streamline API selection (will be removed soon once CORS errors are dealt with)
+   * @param{event} lit event sent by the the text input object within the chat
+   **/
+  _handleAPIselection(event) {
+    console.log(event)
+    const target  = event.target.value;
+    console.log(target)
+    this._selectedapi = target.value;
+  }
+
+  /** send in the latest user message to the api, package it within the messages array and update the DOM
+  **/
+  _sendInput(){
+    let value = this._inputText;
+    this._messageText = "";
+    this._messages.push({"text":value,"origin":this.username,"showButtons":false,"time":this._getCurrentTime()});
+    this._queryInProgress = true;
+    this.requestUpdate();
+    this._updateScroll();
+
+    this.getResults(value).then((res) => {
+      console.log(res)
+      let reply = this._checkForCode(res.reply.split("\nundefined")[0]);
+      if(reply.length == 0 || res.reply == "undefined"){
+        this._messages.push({"text":"Error reaching the server, try again","origin":this.agentname,"showButtons":false,"style":"error","time":this._getCurrentTime()});
+      }else{
+        for(let subreply of reply){
+          this._messages.push({"text":subreply.text,"origin":this.agentname,"showButtons":false,"style":subreply.type,"time":this._getCurrentTime()});
+        }
+      }
+      this._queryInProgress = false;
+      this.requestUpdate();
+      this._updateScroll();
+
+    }).catch(error => {
+      this._messages.push({"text":"Error reaching the model server, try again","origin":this.agentname,"showButtons":false,"style":"error","time":this._getCurrentTime()});
+      this._queryInProgress = false;
+      this.requestUpdate();
+      this._updateScroll();
+    })
+  }
+
+  /** check the returned model response for a specified code delimiter, split and package the string into multiple messages of type 'text' or 'code'
+   * @param{string} string - returned API call response
+   **/
+  _checkForCode(string) {
+    console.log(string);
+    console.log(string.includes('```'));
+    if (!string.includes('```')) {
+      return [{ type: 'text', text: string }];
+    }
+    const strings = string.split('\n');
+    let mode = 'text';
+    // commenting below because I don't see where `previousMode` is being used in the code
```

Comment:
I had added these comments and commented out  `previousMode` variable when resolving the TS errors you ran into in the beginning - can you double check to see if they are needed and are used somewhere?

Reply:
it is! it dictates whether the current parsing mode is "string" aka plain text or "code", aka a code block. since the response contains both I need a value to toggle between both given a predefined delimiter. I really don't know why TS throws errors on this, lmk how to resolve this.

Reply:
TS is throwing the error because you created the variable and set a value, but it's not detecting where in the code you are utilizing the variable. Where in the code are you checking the `previousMode` value?

Reply:
I see you set it with the initial "text" value and then set it to the `mode` value, but is `previousMode` passed into an object somewhere? I'm not seeing where it's used